File: scripts/algorithms/N/Number of Segments in a String/Number of Segments in a String.js
// Runtime: 46 ms (Top 82.9%) | Memory: 41.90 MB (Top 24.93%)

var countSegments = function(s) {
    let count = 0;
    
    for (let i = 0; i < s.length; i++) {
        // Check if the current character is a non-space character and the previous character is a space or the start of the string
        if (s[i] !== ' ' && (i === 0 || s[i - 1] === ' ')) {
        count++;
        }
    }
    
    return count;
}

File: scripts/algorithms/N/Number of Segments in a String/Number of Segments in a String.java
// Runtime: 0 ms (Top 100.00%) | Memory: 41.9 MB (Top 35.26%)
class Solution {
    public int countSegments(String s) {
        int length = 0;
        boolean flag = false;

        for(Character c : s.toCharArray()) {
            if(c == ' ' && flag) {
                length++;
                flag = !flag;
            } else if(c != ' ') {
                flag = true;
            }
        }

        return flag ? length + 1 : length;
    }
}